compiler/tests-common/tests/org/jetbrains/kotlin/codegen/GenerationUtils.kt

```diff
@@ -65,22 +88,69 @@ object GenerationUtils {
         packagePartProvider: (GlobalSearchScope) -> PackagePartProvider,
         trace: BindingTrace = NoScopeRecordCliBindingTrace()
     ): GenerationState {
-        val analysisResult =
-            JvmResolveUtil.analyzeAndCheckForErrors(files.first().project, files, configuration, packagePartProvider, trace)
-        analysisResult.throwIfError()
+        val project = files.first().project
+        val state = if (configuration.getBoolean(CommonConfigurationKeys.USE_FIR)) {
```

Comment:
I think two branches of this `if` are worth extracting to separate functions (will do myself).